snapcraft/internal/project_loader/_templates/_template.py

```diff
@@ -0,0 +1,59 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from typing import Any, Dict, Set
+
+
+class Template:
+    @classmethod
+    def supported_bases(cls) -> Set[str]:
```

Comment:
If we need to add parens, this reads better and leads to less pythonic errors:
```python
if "core18" in <template>.get_supported_bases()
```

If not, use a property:
```python
if "core18" in <template>supported_bases:
```

Or just a class attribute and rely on python's duck typing.

Reply:
I thought about making this a class attribute, but I think using a method makes for a much easier-to-implement API. I find adding "get" actually decreases readability ("if core18 in template get supported bases" doesn't read as well as "if core18 in template supported bases") and be somewhat meaningless (particularly since there's no "set"), but will add it if you like. However, if we add it to this method, we should probably add it to the rest of the methods in the class as well. Thoughts?

Reply:
making it an attribute at least fails fast and in your face, adding get_ makes it sort of obvious you need to add parens as it represents an action.

Reply:
As you requested offline, the `Template` API has been moved entirely to attributes.